cfgov/unprocessed/apps/youth-employment-success/js/views/review/choice.js

```diff
@@ -0,0 +1,118 @@
+import { checkDom, setInitFlag } from '../../../../../js/modules/util/atomic-helpers';
+import { updateRouteChoiceAction } from '../../reducers/choice-reducer';
+import { toArray } from '../../util';
+import transportationMap from '../../data/transportation-map';
+import inputView from '../input';
+
+const CLASSES = Object.freeze( {
+  CONTAINER: 'js-yes-review-choice',
+  REVIEW_PLAN: 'js-yes-plans-review',
+  BUTTON: 'js-yes-choice-finalize',
+  CHOICE: 'js-yes-route-selection'
+} );
+
+function reviewChoiceView( element, { store } ) {
+  const _dom = checkDom( element, CLASSES.CONTAINER );
+  const _reviewPlanEl = _dom.querySelector( `.${ CLASSES.REVIEW_PLAN }` );
+  const _choiceBtnEls = toArray(
+    _dom.querySelectorAll( `.${ CLASSES.CHOICE }` )
+  );
+  const _reviewBtnEl = _dom.querySelector( `.${ CLASSES.BUTTON }` );
+
+  function _enableChoices() {
+    _choiceBtnEls.forEach( el => el.querySelector( 'input' ).setAttribute( 'disabled', false )
+    );
+  }
+
+  function _disableChoices() {
+    _choiceBtnEls.forEach( el => el.querySelector( 'input' ).setAttribute( 'disabled', 'disabled' )
+    );
+  }
+
+  function _enableReviewButton() {
+    _reviewBtnEl.setAttribute( 'disabled', '' );
+  }
+
+  function _disableReviewButton() {
+    _reviewBtnEl.setAttribute( 'disabled', 'disabled' );
+  }
+
+  function _populateOptionLabels( routes ) {
+    _choiceBtnEls.forEach( ( el, index ) => {
+      const route = routes[index];
+
+      // Need a check here because the third option isnt tied to a route.s
+      if ( route ) {
+        const label = el.querySelector( 'label' );
+        const friendlyOption = transportationMap[route.transportation];
+        const nextLabel = label.textContent.replace( '-', friendlyOption );
+        label.textContent = nextLabel;
+      }
+    } );
+  }
+
+  function _showReviewPlan() {
+    _reviewPlanEl.classList.remove( 'u-hidden' );
+    _reviewBtnEl.removeEventListener( 'click', _showReviewPlan );
+  }
+
+  function _hideReviewPlan() {
+    _reviewPlanEl.classList.add( 'u-hidden' );
+  }
+
+  function _handlePlanSelection( { event } ) {
+    store.dispatch(
+      updateRouteChoiceAction( event.target.value )
+    );
+  }
+
+  function _initInputs() {
+    _choiceBtnEls.forEach( el => {
+      inputView( el.querySelector( 'input' ), {
+        events: {
+          click: _handlePlanSelection
+        },
+        type: 'radio'
+      } ).init();
+    } );
+
+    _reviewBtnEl.addEventListener( 'click', _showReviewPlan );
+  }
+
+  function _handleStateUpdate( _, state ) {
+    const transportationSelected = state.routes.routes.reduce( ( _, route ) => ( route && Boolean( route.transportation ) ? true : false ), false );
```

Comment:
Hmmm the logic pathways in this line are pretty convoluted. Could this be broken apart for clarity?

Reply:
The linter weirdly made this harder to read 😦